test/index.js

```diff
@@ -0,0 +1,54 @@
+const { exec } = require('child_process');
+// eslint-disable-next-line node/no-unpublished-require
+const { test } = require('tapzero');
+const package = require('../package.json');
+
+const { devDependencies, version } = package;
+const FAKE = 'FAKE';
+
+test('should not have dependencies from now until forever', async (t) => {
+  const depKey = 'dependencies';
+  const hasDeps = Object.prototype.hasOwnProperty.call(package, depKey);
+  t.equal(hasDeps, false);
+});
+
+test('should only have ten devDependency from now until forever', async (t) => {
+  const { length } = Object.keys(devDependencies);
+  t.equal(length, 10);
+});
+
+test('should not error when performing a dry-run publish', async (t) => {
+  await new Promise((resolve, reject) => {
+    exec('./scripts/publish.sh', (error) => {
+      if (error) {
+        reject(new Error(error));
+      }
+      t.equal(error, null);
+      resolve();
+    });
+  });
+});
+
+test('correct version should appear in dry-run output', async (t) => {
+  await new Promise((resolve, reject) => {
+    exec('./scripts/publish.sh', (error, __, stderr) => {
+      if (error) {
+        reject(new Error(error));
+      }
+      t.equal(stderr.includes(version), true);
+      resolve();
+    });
+  });
+});
+
+test('should error when token is invalid', async (t) => {
+  await new Promise((resolve, reject) => {
+    exec(`NPM_TOKEN=${FAKE} ./scripts/publish.sh`, (error) => {
+      if (!error) {
+        reject(new Error('should error when token is invalid'));
+      }
+      t.equal(error.code, 1);
```

Comment:
Testing the exit code for `main.sh` and not just `publish.sh` is important because we can make sure that the action is reported as a failure if something goes wrong during publishing.

Reply:
updated to test `main.sh` instead